src/roswire/common/package.py

```diff
@@ -12,18 +15,85 @@
     List,
     Mapping,
     Optional,
+    Set,
 )
 
 import attr
 from loguru import logger
 
+from .cmake import ParserContext as CMakeParserContext, argparse as cmake_argparse
 from .action import ActionFormat
 from .msg import MsgFormat
 from .srv import SrvFormat
+from ..util import tuple_from_iterable
+
 
 if typing.TYPE_CHECKING:
     from .. import AppInstance
 
+
+class SourceLanguage(enum.Enum):
+    CXX = "cxx"
+    PYTHON = "python"
+
+
+@attr.s(frozen=True, auto_attribs=True, slots=True)
+class NodeSourceInfo:
+    node_name: Optional[str]
```

Comment:
There shouldn't be a case where a node doesn't have a name right?

Reply:
Done.